### ISSUE TITLE
New MISSING_VAR false positives in 1.3.0 around cfscript/tag syntax

### ISSUE THREAD

**Issue by TheRealAgentK** [cflint/CFLint#517]
ACF, here are a few examples I could easily find, there might be tons more. 

These never triggered false positives in 1.2.x.

```var qGetSites = new Query(datasource="...", sql="...");```
-> Variable datasource is not declared with a var statement, same for "variable" sql.

```var mailService=new mail(
                    to="...",
                    from="...",
                    subject="...",
                    type="HTML",
                    body=mailBody
                    );```
-> all 5 valid attributes here trigger now MISSING_VAR

```  var httpRequest = new HTTP(
        method = "get",
        url = ("http://api.pusherapp.com" & resourceUri),
        charset = "utf-8"
            );```
-> method and charset trigger a MISSING_VAR, url doesn't. No idea why.

```var httpRequest = new Http(method=arguments.method, url=variables.ENDPOINT & arguments.resourceUri, charset="utf-8", timeout=arguments.timeout, throwOnError=false, username=variables.something, password=variables.somethingelse, getAsBinary="never");```
-> charset, throwOnError, username, password, getAsBinary trigger MISSING_VAR

**Comment by TheRealAgentK** [comment]
I remember having seen a ticket that dealt with those built-in CFML objects for 1.3.0 - any chance whoever worked on it could have a look into these issues?

**Comment by KamasamaK** [comment]
There was #482 and #483 which seem similar.

**Comment by TheRealAgentK** [comment]
Yeah, it seems that something related to the fixes for either of these tickets now creates the issues with the "new" syntax.

**Comment by ryaneberly** [comment]
Sounds like the improved parsing is causing more false positives.

**Comment by timbeadle** [comment]
👍 I've seen this issue as well.

**Comment by ryaneberly** [comment]
scopes ref:
https://help.adobe.com/en_US/ColdFusion/9.0/Developing/WSc3ff6d0ea77859461172e0811cbec09af4-7ff1.html

**Comment by ryaneberly** [comment]
@TheRealAgentK, are you getting the false positives on the dev branch?  I'm not aware of any fixes, but the tests I added to
https://github.com/cflint/CFLint/commit/2b4a9aa759bc155e7cac9c6690d61e504224a6fb
based on your examples do not generate false positives.

**Comment by TheRealAgentK** [comment]
Pretty sure I got the issue with a local build of the dev branch. Your tests look like my scenarios - I'll look closer into that later today or on the weekend - weird.

**Comment by lmajano** [comment]
Yep, this is 1.3.0 and here are some more code samples if needed:

```
var oExpectation = new Expectation( spec=this, assertions=this.$assert, mockbox=this.$mockbox );

// Interesting one
thread.closures = arguments.closures;

```

**Comment by ryaneberly** [comment]
Hi @lmajano 
Can you provide a file example.   I do not get the problem on:

    component{
      function foo(){
        var oExpectation = new Expectation( spec=this, assertions=this.$assert, mockbox=this.$mockbox );
        thread.closures = arguments.closures;
      } 
    }

**Comment by KamasamaK** [comment]
@ryaneberly I am able to reproduce the issue on 1.3.0 as it is. However, it appears to be fine on the latest `dev` snapshot.

**Comment by ryaneberly** [comment]
Interesting.  I downloaded the 1.3.0 from maven and couldn't reproduce it in isolation.  Thanks for checking.

**Comment by ryaneberly** [comment]
I think this can be closed and included in 1.4.0
